pkg/process/metadata/parser/java/util.go

```diff
@@ -0,0 +1,126 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rickar/props"
+	"github.com/vibrantbyte/go-antpath/antpath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type mapSource struct {
+	m map[string]string
+}
+
+// Get the value for the supplied key
+func (y *mapSource) Get(key string) (string, bool) {
+	val, ok := y.m[key]
+	return val, ok
+}
+
+// GetDefault gets the value for the supplied key or the defVal if missing
+func (y *mapSource) GetDefault(key string, defVal string) string {
+	val, ok := y.m[key]
+	if !ok {
+		return defVal
+	}
+	return val
+}
+
+func newArgumentSource(arguments []string, prefix string) *mapSource {
+	parsed := make(map[string]string)
+	for _, val := range arguments {
+		if strings.HasPrefix(val, prefix) {
+			parts := strings.SplitN(val[len(prefix):], "=", 2)
+			if len(parts) == 1 {
+				parsed[parts[0]] = ""
+			} else {
+				parsed[parts[0]] = parts[1]
+			}
+			continue
+		}
+	}
+	return &mapSource{parsed}
+}
+
+func newPropertySourceFromStream(rc io.ReadCloser, filename string) (*props.PropertyGetter, error) {
```

Comment:
what about case sensitive? having YAML? YML?

Reply:
that's a great question and definitely yes the file type switch should not be case sensitive. Even if is not common to have MYFILE.YAML, this can likely happen (especially on windows) and especially when the user is explicitly providing an absolute file. I will change to address this thank you